internal/drivers/trivial/auth.go

```diff
@@ -0,0 +1,143 @@
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+******************************************************************************/
+
+package trivial
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+
+	"github.com/sapcc/keppel/internal/keppel"
+
+	"github.com/go-redis/redis"
+	"github.com/sapcc/go-bits/audittools"
+	"github.com/sapcc/go-bits/logg"
+)
+
+type AuthDriver struct{}
+
+func init() {
+	keppel.RegisterUserIdentity("trivial", deserializeTrivialUserIdentity)
+	keppel.RegisterAuthDriver("trivial", func(rc *redis.Client) (keppel.AuthDriver, error) { return &AuthDriver{}, nil })
+}
+
+func deserializeTrivialUserIdentity(in []byte, _ keppel.AuthDriver) (keppel.UserIdentity, error) {
+	var uid userIdentity
+	err := json.Unmarshal(in, &uid)
+	return uid, err
+}
+
+type userIdentity struct {
+	Username string
+}
+
+func (uid userIdentity) HasPermission(perm keppel.Permission, tenantID string) bool {
+	return true
+}
+
+func getUserName() string {
+	userName := os.Getenv("KEPPEL_USERNAME")
+	if userName == "" {
+		logg.Fatal("KEPPEL_USERNAME env cannot be empty when using trivial auth")
+	}
+	return userName
+}
+
+type DummyUserInfo struct{}
+
+func (DummyUserInfo) UserUUID() string {
+	return ""
+}
+
+func (DummyUserInfo) UserName() string {
+	return getUserName()
+}
+
+func (DummyUserInfo) UserDomainName() string {
+	return ""
+}
+
+func (DummyUserInfo) ProjectScopeUUID() string {
+	return ""
+}
+
+func (DummyUserInfo) ProjectScopeName() string {
+	return ""
+}
+
+func (DummyUserInfo) ProjectScopeDomainName() string {
+	return ""
+}
+
+func (DummyUserInfo) DomainScopeUUID() string {
+	return ""
+}
+
+func (DummyUserInfo) DomainScopeName() string {
+	return ""
+}
+
+func (uid userIdentity) UserInfo() audittools.UserInfo {
+	return DummyUserInfo{}
+}
+
+func (uid userIdentity) UserName() string {
+	return getUserName()
+}
+
+func (uid userIdentity) UserType() keppel.UserType {
+	return keppel.RegularUser
+}
+
+func (uid userIdentity) SerializeToJSON() (typeName string, payload []byte, err error) {
+	payload, err = json.Marshal(uid)
+	return "trivial", payload, err
+}
+
+func (d *AuthDriver) AuthenticateUser(userName, password string) (keppel.UserIdentity, *keppel.RegistryV2Error) {
+	expectedUserName := os.Getenv("KEPPEL_USERNAME")
+	if expectedUserName == "" {
+		return nil, keppel.ErrUnauthorized.With(`KEPPEL_USERNAME env cannot be empty when using trivial auth`)
+	}
+	expectedPassword := os.Getenv("KEPPEL_PASSWORD")
+	if expectedPassword == "" {
+		return nil, keppel.ErrUnauthorized.With(`KEPPEL_PASSWORD env cannot be empty when using trivial auth`)
+	}
```

Comment:
Since environment variables do not usually change after a program has started, I consider it better style to read them only once during initialization, as do all the other drivers that take configuration from environment variables. This would also avoid the code duplication that currently exists between the different methods of this AuthDriver.